q2_demux/_filter.py

```diff
@@ -0,0 +1,42 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2019, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import pandas as pd
+
+from qiime2 import Metadata
+from qiime2.util import duplicate
+from q2_demux._summarize import _PlotQualView
+from q2_types.per_sample_sequences import \
+    CasavaOneEightSingleLanePerSampleDirFmt
+
+
+def filter_samples(samples: _PlotQualView, metadata: Metadata,
+                   where: str = None, exclude_ids: bool = False) \
+                   -> CasavaOneEightSingleLanePerSampleDirFmt:
+    dir = CasavaOneEightSingleLanePerSampleDirFmt()
+
+    paired = samples.paired
+    samples = samples.directory_format
+
+    ids_to_keep = metadata.get_ids(where=where)
+    manifest = samples.manifest.view(pd.DataFrame)
+
+    if exclude_ids:
+        ids_to_keep = set(manifest.index) - set(ids_to_keep)
+
+    try:
+        for id in ids_to_keep:
+            forward = manifest.loc[id].forward
+            duplicate(forward, str(dir) + '/' + forward.rsplit('/', 1)[1])
```

Comment:
use `os.path` operations here instead of relying on `/` to be the separator. There is a function to get your the filename as well.